observability/akka-http-metrics/src/test/suite/scala/com/daml/metrics/akkahttp/AkkaHttpMetricsSpec.scala

```diff
@@ -0,0 +1,621 @@
+          metrics.httpRequestsTotalValue should be(3)
+        }
+      }
+    }
+
+    "collect missing routes" in {
+      withRouteAndMetrics { (route, metrics) =>
+        Get("/undefined") ~> route
+        Get("/otherUndefined") ~> route ~> check {
+          metrics.httpRequestsTotalValue should be(2)
+        }
+      }
+    }
+
+    "collect unauthorized requests" in {
+      withRouteAndMetrics { (route, metrics) =>
+        Get("/unauthorized") ~> route ~> check {
+          metrics.httpRequestsTotalValue should be(1)
+        }
+      }
+    }
+
+    "collect bad requests" in {
+      withRouteAndMetrics { (route, metrics) =>
+        Get("/badrequest") ~> route ~> check {
+          metrics.httpRequestsTotalValue should be(1)
+        }
+      }
+    }
+
+    "collect requests resulting in exceptions" in {
+      withRouteAndMetrics { (route, metrics) =>
+        Get("/exception") ~> route ~> check {
+          metrics.httpRequestsTotalValue should be(1)
+        }
+      }
+    }
+
+  }
+
+  "errors_total" should {
+    "collect missing routes" in {
+      withRouteAndMetrics { (route, metrics) =>
+        Get("/undefined") ~> route
+        Get("/otherUndefined") ~> route ~> check {
+          metrics.httpErrorsTotalValue should be(2)
+        }
+      }
+    }
+
+    "collect unauthorized requests" in {
+      withRouteAndMetrics { (route, metrics) =>
+        Get("/unauthorized") ~> route ~> check {
+          metrics.httpErrorsTotalValue should be(1)
+        }
+      }
+    }
+
+    "collect bad requests" in {
+      withRouteAndMetrics { (route, metrics) =>
+        Get("/badrequest") ~> route ~> check {
+          metrics.httpErrorsTotalValue should be(1)
+        }
+      }
+    }
+
+    "collect requests resulting in exceptions" in {
+      withRouteAndMetrics { (route, metrics) =>
+        Get("/exception") ~> route ~> check {
+          metrics.httpErrorsTotalValue should be(1)
+        }
+      }
+    }
+
+    "not collect successful requests" in {
+      withRouteAndMetrics { (route, metrics) =>
+        Get() ~> route
+        Get("/simple") ~> route
+        // needs one failing request, otherwise no value can be found for the metric
+        Get("/undefined") ~> route
+        Get("/a/bit/deeper") ~> route ~> check {
+          metrics.httpErrorsTotalValue should be(1)
+        }
+      }
+    }
+
+  }
+
+  "requests_bytes_total" should {
+    "record successful request without payload" in {
+      withRouteAndMetrics { (route, metrics) =>
+        Get("/a/bit/deeper") ~> route ~> check {
+          metrics.httpRequestsBytesTotalValue should be(0L)
+        }
+      }
+    }
+
+    "record string payload size on successful request" in {
+      withRouteAndMetrics { (route, metrics) =>
+        Get(
+          "/mirror/200",
+          HttpEntity.Strict(ContentTypes.`text/plain(UTF-8)`, ByteString("hello")),
+        ) ~> route ~> check {
+          metrics.httpRequestsBytesTotalValue should be(5L)
```

Comment:
To improve readability, I'd suggest explicitly calling `.size()` on the used `ByteString` objects instead:
```scala
val helloBytes = ByteString("hello")
...
  metrics.httpRequestsBytesTotalValue should be(helloBytes.size())
```